collator/src/mempool/impls/std_impl/cache.rs

```diff
@@ -55,20 +69,23 @@ impl Cache {
                             target: tracing_targets::MEMPOOL_ADAPTER,
                             %anchor_id,
                             %first_id,
+                            is_paused = Some(data.is_paused).filter(|x| *x),
                             "Requested anchor is too old"
                         );
                         return None;
                     }
                     _ => {
-                        let found = self.anchors.read().get(&anchor_id).cloned();
-                        if found.is_none() {
-                            tracing::warn!(
-                                target: tracing_targets::MEMPOOL_ADAPTER,
-                                %anchor_id,
-                                "Anchor is unknown"
-                            );
-                        }
-                        return found;
+                        match data.anchors.get(&anchor_id) {
+                            Some(found) => return Some(found.clone()),
+                            None => {
+                                tracing::warn!(
+                                    target: tracing_targets::MEMPOOL_ADAPTER,
+                                    %anchor_id,
+                                    is_paused = Some(data.is_paused).filter(|x| *x),
+                                    "Anchor is unknown, waiting"
+                                );
+                            }
```

Comment:
looks like this is the only change in ``Cache` behavior: if mempool is expected to be not paused, then collator will wait for anchor to be received.